src/services/TestResultsService.ts

```diff
@@ -491,10 +491,18 @@ export class TestResultsService {
     return bool;
   }
 
+  public isHGVTRLRoadworthinessTest(testTypeId: string): boolean {
+    return Object.values(HGV_TRL_ROADWORTHINESS_TEST_TYPES).includes(testTypeId);
+   }
+
   public generateCertificateNumber(payload: ITestResultPayload) {
     if (payload.testStatus === TEST_STATUS.SUBMITTED) {
       payload.testTypes.forEach((testType) => {
-        if (testType.testTypeClassification === TEST_TYPE_CLASSIFICATION.ANNUAL_WITH_CERTIFICATE && testType.testResult !== TEST_RESULT.ABANDONED && !this.isTestTypeAdr(testType) && !this.isTestTypeLec(testType)) {
+        // CVSB-7675 if vehicle type is HGV/TRL and testTypeId is Roadworthiness test and testResult is pass then testNumber = certificateNumber
+       if ( (payload.vehicleType === VEHICLE_TYPES.HGV || payload.vehicleType === VEHICLE_TYPES.TRL && this.isHGVTRLRoadworthinessTest(testType.testTypeId) && testType.testResult === TEST_RESULT.PASS)
```

Comment:
Getting extremely unwieldy. I'd be inclined to split this whole thing out to a Util function with some more named functions, just to make the logic easier to read e.g. isHGVorTRLVehicle(payload),